packages/http/src/types.ts

```diff
@@ -510,6 +518,28 @@ export interface HttpOperationPart {
   readonly headers: HeaderProperty[];
   /** If there can be multiple of that part */
   readonly multi: boolean;
+  /** The part name, if any. */
+  readonly name?: string;
+  /** If the part is optional */
+  readonly optional: boolean;
+}
+
+export type HttpOperationPart = HttpOperationModelPart | HttpOperationTuplePart;
+
+/** Represents a part in a multipart body that comes from a model property. */
+export interface HttpOperationModelPart extends HttpOperationPartCommon {
+  readonly partKind: "model";
+  /** Property that defined the part if the model form is used. */
+  readonly property: ModelProperty;
+  /** Part name */
+  readonly name: string;
+}
+
+/** Represents a part in a multipart body that comes from a tuple entry. */
+export interface HttpOperationTuplePart extends HttpOperationPartCommon {
+  readonly partKind: "tuple";
+  /** Property that defined the part -- always undefined for tuple entry parts. */
+  readonly property?: undefined;
```

Comment:
we have to do this so you can still reference `property` without checking the partkind?

Reply:
Yes, but technically it also restricts what `property` can be in the interface. Logically in a structural type system the constraint of `property` in an interface that doesn't specify it is `unknown` and you can do some things in TypeScript that let you utilize that openness of the constraint (like spreads, transitive assignments, etc.). Strict checking of properties in object literals and accesses stops you from doing `.property` or specifying `property` in an object _literal_, but specifying the constraint `: unknown` prevents a lot of invalid constructions that are otherwise allowed by the structural type system in a lot of cases.